packages/sign-client/src/controllers/engine.ts

```diff
@@ -856,6 +866,12 @@ export class Engine extends IEngine {
     }
     const { pairingTopic, requiredNamespaces, relays } = params;
     if (!isUndefined(pairingTopic)) await this.isValidPairingTopic(pairingTopic);
+
+    // validate required namespaces only if they are defined
+    if (!isUndefined(requiredNamespaces) && isValidObject(requiredNamespaces) === 0) {
```

Comment:
Sorry but I'm confused by the `=== 0` on what should be a boolean return. Can we fix `isValidObject` if it's returning a number? Probably comes from treating a number like a boolean in a boolean concatenation.

`@walletconnect/utils` is also monorepo so shouldn't cause any misalignment issues in terms of versioning.